### PR TITLE
feat: Add types for to-regex-range

### PR DESCRIPTION
**Pull request by Richienb**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/to-regex-range/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    interface Options {
+        capture?: boolean;
+        shorthand?: boolean;
+        relaxZeros?: boolean;
+    }
+}
+
+declare function toRegexRange(
+    min: number | string,
+    max?: number | string,
```
Comment (peterblazejewicz) [inline types/to-regex-range/index.d.ts:16]: should the `max` be either undefined, or when present, a `number`?
https://github.com/micromatch/to-regex-range/blob/master/index.js#L21

Comment (Richienb) [reply types/to-regex-range/index.d.ts:16]: Stringified numbers are also allowed: https://github.com/jonschlinkert/is-number/blob/master/index.js#L14-L16

Comment (peterblazejewicz) [reply types/to-regex-range/index.d.ts:16]: oh, cool, you're right, I've done only first part :)

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@Richienb thx!

**Comment by Richienb** [comment]
Ready to merge

**Pull request merged**